common/remote/rmtsmtp.cpp

```diff
@@ -633,6 +648,74 @@ class CMailInfo
         out.append("HELO ").append(mailServer.get()).append("\r\n");
     }
 
+    void getEhlo(StringBuffer & out) const
+    {
+        out.append("EHLO ").append(mailServer.get()).append("\r\n");
+    }
+
+    void getEhloAndOptions(StringBuffer & out)
+    {
+        // Initial EHLO and gather reply, which should contain ESMTP options
+        getEhlo(out);
```

Comment:
Does this need to fall back to HELO if EHLO not supported?  Can we ignore for any modern mail server?

Reply:
ESMTP came out in 1995 (RFC 1869). All mail servers should support ESMTP by now. That said, a misconfiguration on the server side could prompt a fallback to the original protocol. I will add that.